libimage/pull.go

```diff
@@ -91,36 +89,18 @@ func (r *Runtime) Pull(ctx context.Context, name string, pullPolicy config.PullP
 	switch ref.Transport().Name() {
 
 	// DOCKER REGISTRY
-	case dockerRegistryTransport.Transport.Name():
+	case registryTransport.Transport.Name():
 		pulledImages, pullError = r.copyFromRegistry(ctx, ref, strings.TrimPrefix(name, "docker://"), pullPolicy, options)
 
-	// DOCKER DAEMON
-	case dockerDaemonTransport.Transport.Name():
-		pulledImages, pullError = r.copyFromDefault(ctx, ref, &options.CopyOptions)
-
 	// DOCKER ARCHIVE
 	case dockerArchiveTransport.Transport.Name():
 		pulledImages, pullError = r.copyFromDockerArchive(ctx, ref, &options.CopyOptions)
 
-	// OCI
-	case ociTransport.Transport.Name():
-		pulledImages, pullError = r.copyFromDefault(ctx, ref, &options.CopyOptions)
-
-	// TARBALL
-	case tarballTransport.Transport.Name():
-		pulledImages, pullError = r.copyFromDefault(ctx, ref, &options.CopyOptions)
-
-	// OCI ARCHIVE
-	case ociArchiveTransport.Transport.Name():
-		pulledImages, pullError = r.copyFromDefault(ctx, ref, &options.CopyOptions)
-
-	// DIR
 	case dirTransport.Transport.Name():
-		pulledImages, pullError = r.copyFromDefault(ctx, ref, &options.CopyOptions)
 
-	// UNSUPPORTED
+	// ALL OTHER TRANSPORTS
 	default:
-		return nil, errors.Errorf("unsupported transport %q for pulling", ref.Transport().Name())
+		pulledImages, pullError = r.copyFromDefault(ctx, ref, &options.CopyOptions)
```

Comment:
This makes my spidey sense tingle.  I'd feel better if we still specified the transports that we support and keep the error just in case one we don't support in the future ever sneaks through.

Reply:
containers/image takes care of that already